js/util/string_number_mapping.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = StringNumberMapping;
+
+function StringNumberMapping(strings) {
+    this.stringToNumber = {};
+    this.numberToString = [];
+    for (var i = 0; i < strings.length; i++) {
+        var string = strings[i];
+        this.stringToNumber[string] = i;
+        this.numberToString[i] = string;
+    }
+}
```

Comment:
It's used for converting strings to numbers and back to strings. Kindof like a runtime enum.

It's used for one thing:
- converting a feature's sourceLayer's name into a number
- storing that number in a StructArray
- and then turning that number back into a string so that it can get that vectortile layer.